common/stream.h

```diff
@@ -553,173 +623,209 @@ class ReadStream : virtual public Stream {
 	 *
-	 * @note This methods is closely modeled after the standard fgets()
+	 * @note This method is closely modeled after the standard fgets()
 	 *       function from stdio.h.
 	 *
-	 * @param s	the buffer to store into
-	 * @param bufSize	the size of the buffer
-	 * @param handleCR	if set (default), then CR and CR/LF are handled as well as LF
-	 * @return a pointer to the read string, or NULL if an error occurred
+	 * @param s	t       The buffer to store into.
+	 * @param bufSize	Size of the buffer.
+	 * @param handleCR	If set (default), then CR and CR/LF are handled, as well as LF.
+	 *
+	 * @return Pointer to the read string, or NULL if an error occurred.
 	 */
 	virtual char *readLine(char *s, size_t bufSize, bool handleCR = true);
 
 
 	/**
-	 * Reads a full line and returns it as a Common::String. Reading
-	 * stops when the end of a line is reached (CR, CR/LF or LF), and
-	 * at end-of-file or error.
+	 * Read a full line and returns it as a Common::String.
+	 *
+	 * Reading stops when the end of a line is reached (CR, CR/LF, or LF),
+	 * and at end-of-file or error.
 	 *
 	 * Upon successful completion, return a string with the content
 	 * of the line, *without* the end of a line marker. This method
 	 * does not indicate whether an error occurred. Callers must use
 	 * err() or eos() to determine whether an exception occurred.
 	 *
-	 * @param handleCR	if set (default), then CR and CR/LF are handled as well as LF
+	 * @param handleCR	If set (default), then CR and CR/LF are handled, as well as LF.
 	 */
 	virtual String readLine(bool handleCR = true);
 
 	/**
 	 * Print a hexdump of the stream while maintaing position. The number
 	 * of bytes per line is customizable.
-	 * @param len	the length of that data
-	 * @param bytesPerLine	number of bytes to print per line (default: 16)
-	 * @param startOffset	shift the shown offsets by the starting offset (default: 0)
+	 *
+	 * @param len           Length of this data.
+	 * @param bytesPerLine	Number of bytes to print per line (default: 16).
+	 * @param startOffset	Shift the shown offsets by the starting offset (default: 0).
 	 */
 	void hexdump(int len, int bytesPerLine = 16, int startOffset = 0);
 };
 
 /**
- * This is a ReadStream mixin subclass which adds non-endian read
+ * ReadStream mixin subclass that adds non-endian read
  * methods whose endianness is set during the stream creation.
  */
 class ReadStreamEndian : virtual public ReadStream {
 private:
 	const bool _bigEndian;
 
 public:
+	/**
+	 * Set the endianness of the read stream.
+	 *
+	 * @param bigEndian If set, create a big endian stream.
+	 *					If not set, create a little endian stream.
+	 */
 	ReadStreamEndian(bool bigEndian) : _bigEndian(bigEndian) {}
-
+	/**
+	 * Return true if data is encoded in big endian order.
+	 */
 	bool isBE() const { return _bigEndian; }
-
+	/**
+	 * Read an unsigned 16-bit word and return it
+	 * encoded in big endian or little endian order.
+	 */
```

Comment:
I am commenting on the first relevant case, but this applies to all the comments in this class.
Here the idea is to read a value using the stream endianness and return it in native endianness. So for example if we create the stream with `bigEndian = true`, then `readUint16()` does the same thing as `readUint16BE()` (that is inherited from `ReadStream`): it reads a 16 bit unsigned integer big endian value and convert it to the native endianness.

Here are some details
The code here is confusing, and I think it was meant to use
```C++
return (_bigEndian) ? FROM_BE_16(val) : FROM_LE_16(val);
```

The behaviour of `TO_BE_16` is exactly the same as `FROM_BE_16`: if we are on a big endian system they both do nothing, and if we are on a little endian system they both swap the bytes.

But semantically the current implementation seems to say "I am reading from a stream that uses native endianness and I want a big endian value". And it would make more sense to say "I am reading from a big endian stream and I want the value in native endianness.

That is because often the streams are the game data file, and we know the endianness of those files (usually this is the endianness of the system on which the game was released, so for example BE for mac games and LE for DOS games). And we read those and want to convert them to the native endianness.

And you can see this logic in the ReadStream implementation. For example:
```C++
uint16 readUint16BE() {
	uint16 val;
	read(&val, 2);
	return FROM_BE_16(val);
}
```
I think here it was meant to do the same, and in actual fact the original implementation was:
```C++
inline uint16 readUint16() {
	return (_bigEndian) ? readUint16BE() : readUint16LE();
}
```
This was changed in a big refactoring to the current code, and I believe the use of `TO_*` instead of `FROM_*` was a mistake. I will change those, but maybe wait until after this PR is merged to avoid conflicts.

Reply:
Ok, so right now I'm changing this (and other occurrences respectively) to:
``
	 * Read an unsigned 16-bit word using the stream endianness
	 * and return it in native endianness.
``
and then this can be changed when you make the fix.
Nice to see we can also make some improvements in the code as part of this project.